### PR TITLE
Custom serializer register

### PR DESCRIPTION
**Pull request by Lo1nt**
### Motivation:

Currently, the custom serializer is registered by static code block. And once it registered, it could not be modified.

When there are several classes that register custom serializer, the sequence, dependents on sequence of class loading, is unpredictable, and unmodifiable. 

Thus, there's no way we define our own serializer (say, for SofaRequest.Class).

The pr here is aiming at making custom serializer registration sequentially.

P.S. And we here still don't want to change the unmodifiable feature (Once a serializer registered, it can never be changed).

### Modification:

Apply extension-loader mechanism to let the register work in sequence.

Still, there is an override feature for extension loader, which enables us to substitute the origin register provided by sofa rpc. 

### Result:

1. Let all register become instance managed by ExtensionLoader.
2. Change call to static register method to get Extension instance and then call instances' register method.

### CODE REVIEW

File: remoting/remoting-bolt/src/main/java/com/alipay/sofa/rpc/server/bolt/BoltServerProcessor.java [hunk 1]
```diff
@@ -69,7 +72,10 @@ public class BoltServerProcessor extends AsyncUserProcessor<SofaRequest> {
      * 提前注册序列化器
      */
     static {
-        SofaRpcSerializationRegister.registerCustomSerializer();
+        String extensionAlias = RpcConfigs.getOrDefaultValue(RpcOptions.BOLT_SERIALIZER_REGISTER_EXTENSION,
```
Comment (OrezzerO) [inline remoting/remoting-bolt/src/main/java/com/alipay/sofa/rpc/server/bolt/BoltServerProcessor.java:75]: If these static code can remove into constructor?

File: remoting/remoting-bolt/src/main/resources/META-INF/services/sofa-rpc/com.alipay.sofa.rpc.codec.bolt.AbstractSerializationRegister [hunk 2]
```diff
@@ -0,0 +1 @@
+sofaRpcSerializationRegister=com.alipay.sofa.rpc.codec.bolt.SofaRpcSerializationRegister
```
Comment (OrezzerO) [inline remoting/remoting-bolt/src/main/resources/META-INF/services/sofa-rpc/com.alipay.sofa.rpc.codec.bolt.AbstractSerializationRegister:1]: Add empty line at the end of file.

File: remoting/remoting-bolt/src/test/java/com/alipay/sofa/rpc/codec/bolt/BoltSerializationInitTest.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        Assert.assertNotNull("testRegister未能覆盖原版register", CustomSerializerManager.getCustomSerializer(
+            TestSofaRpcSerializationRegister.class.getName()));
+    }
+
+    @After
+    public void clearClassSerializerMap() {
+        CustomSerializerManager.clear();
+    }
+
+}
```
Comment (OrezzerO) [inline remoting/remoting-bolt/src/test/java/com/alipay/sofa/rpc/codec/bolt/BoltSerializationInitTest.java:52]: Add empty line at the end of file.

File: remoting/remoting-bolt/src/test/resources/META-INF/services/com.alipay.sofa.rpc.codec.bolt.AbstractSerializationRegister [hunk 4]
```diff
@@ -0,0 +1 @@
+sofaRpcSerializationRegister=com.alipay.sofa.rpc.test.TestSofaRpcSerializationRegister
```
Comment (OrezzerO) [inline remoting/remoting-bolt/src/test/resources/META-INF/services/com.alipay.sofa.rpc.codec.bolt.AbstractSerializationRegister:1]: Add empty line at the end of file.

File: remoting/remoting-bolt/src/test/java/com/alipay/sofa/rpc/test/TestSofaRpcSerializationRegister.java [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * Register custom serializer to bolt.
+ *
+ * @author <a href="mailto:zhanggeng.zg@antfin.com">GengZhang</a>
+ */
+@Extension(value = "sofaRpcSerializationRegister", override = true, order = 20)
+public class TestSofaRpcSerializationRegister extends AbstractSerializationRegister {
```
Comment (OrezzerO) [inline remoting/remoting-bolt/src/test/java/com/alipay/sofa/rpc/test/TestSofaRpcSerializationRegister.java:33]: Remove unused code of this class.

File: remoting/remoting-bolt/src/test/java/com/alipay/sofa/rpc/codec/bolt/BoltSerializationInitTest.java [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+/**
+ *
+ * @author junyuan
+ * @version BoltSerializationInitTest.java, v 0.1 2022年12月23日 16:26 junyuan Exp $
+ */
+public class BoltSerializationInitTest {
+
+    @Test
+    public void SerializerRegisterOverrideTest() {
```
Comment (EvenLjj) [inline remoting/remoting-bolt/src/test/java/com/alipay/sofa/rpc/codec/bolt/BoltSerializationInitTest.java:35]: Changing the name of the single test method to testSerializerRegisterOverride is more appropriate

**Review by EvenLjj — APPROVED** [review]
LGTM

**Review by chuailiwu — APPROVED** [review]
lgtm

**Review by leslieSll — APPROVED** [review]
LGTM

**Pull request merged**